src/Features/Core/Portable/Completion/Providers/Snippets/ISnippetService.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers.Snippets
+{
+    internal interface ISnippetService : ILanguageService
+    {
+        Task<ImmutableArray<SnippetData?>> GetSnippetsAsync(Document document, int position, CancellationToken cancellationToken);
+        ISnippetProvider GetSnippetProvider(SnippetData data);
+        Task<TextSpan> GetInvocationSpanAsync(Document document, int position, CancellationToken cancellationToken);
+    }
```

Comment:
most important thing before reviewing is that this be extremely well documented as to what all this means (potentially with examples).  can you do that first?